src/Lean/Syntax.lean

```diff
@@ -530,4 +530,27 @@ def Stack.matches (stack : Syntax.Stack) (pattern : List $ Option SyntaxNodeKind
     |>.all id)
 
 end Syntax
+
+namespace TSyntax
+
+/--
+Applies the given function to every subsyntax.
+
+Like `Syntax.replaceM` but for typed syntax.
+(Note there are no guarantees of type correctness here.)
```

Comment:
I think I'd prefer the abstraction to just not exist then, unsafely creating `TSyntax` should be explicit

Reply:
Turns out it is not even used. I will remove from this PR and maybe clean it up in Std later.